codec/src/main/java/io/netty/handler/codec/DefaultTextHeaders.java

```diff
@@ -396,4 +404,237 @@ public TextHeaders clear() {
+        }
+
+        private TextHeaders addConverted(CharSequence name, CharSequence convertedValue) {
+            CharSequence currentValue = DefaultTextHeaders.super.get(name);
+            if (currentValue == null) {
+                DefaultTextHeaders.super.add(name, convertedValue);
+            } else {
+                DefaultTextHeaders.super.set(name, commaSeparate(currentValue, convertedValue));
+            }
+            return DefaultTextHeaders.this;
+        }
+
+        private CharSequence convertAndCommaSeparate(Object... values) {
+            StringBuilder singleValue = new StringBuilder(values.length * DEFAULT_VALUE_SIZE);
+            if (values.length > 0) {
+                int end = values.length - 1;
+                for (int i = 0; i < end; i++) {
+                    convertAndQuote(singleValue, values[i]).append(", ");
+                }
+                convertAndQuote(singleValue, values[end]);
+            }
+            return singleValue.toString();
+        }
+
+        private CharSequence commaSeparate(CharSequence[] values) {
+            StringBuilder singleValue = new StringBuilder(values.length * DEFAULT_VALUE_SIZE);
+            if (values.length > 0) {
+                int end = values.length - 1;
+                for (int i = 0; i < end; i++) {
+                    quote(singleValue, values[i]).append(", ");
+                }
+                quote(singleValue, values[end]);
+            }
+            return singleValue.toString();
+        }
+
+        private CharSequence convertAndCommaSeparate(Iterable<?> values) {
+            StringBuilder sb = new StringBuilder();
+            Iterator<?> iterator = values.iterator();
+            if (iterator.hasNext()) {
+                Object next = iterator.next();
+                while (iterator.hasNext()) {
+                    convertAndQuote(sb, next).append(", ");
+                    next = iterator.next();
+                }
+                convertAndQuote(sb, next);
+            }
+            return sb.toString();
+        }
+
+        private CharSequence commaSeparate(Iterable<? extends CharSequence> values) {
+            StringBuilder sb = new StringBuilder();
+            Iterator<? extends CharSequence> iterator = values.iterator();
+            if (iterator.hasNext()) {
+                CharSequence next = iterator.next();
+                while (iterator.hasNext()) {
+                    quote(sb, next).append(", ");
+                    next = iterator.next();
+                }
+                quote(sb, next);
+            }
+            return sb.toString();
+        }
+
+        private StringBuilder convertAndQuote(StringBuilder sb, Object value) {
+            return quote(sb, valueConverter.convertObject(value));
+        }
+
+        private StringBuilder quote(StringBuilder sb, CharSequence value) {
+            String converted = value.toString();
+            if (converted.isEmpty()) {
```

Comment:
nit: I'm not sure you need this empty check.  if `converted.contains(COMMA)` is true then the string will not be empty and `converted.charAt(0)` is OK.